File: app/src/main/java/com/esoxjem/movieguide/details/DetailsComponent.java
package com.esoxjem.movieguide.details;

import dagger.Subcomponent;

/**
 * @author arunsasidharan
 */
@DetailsScope
@Subcomponent(modules = {DetailsModule.class})
public interface DetailsComponent {
    void inject(MovieDetailsFragment target);
}

File: app/src/main/java/com/esoxjem/movieguide/movies/details/DetailsComponent.kt
package com.esoxjem.movieguide.movies.details

import dagger.Subcomponent

/**
 * @author arunsasidharan
 */
@DetailsScope
@Subcomponent(modules = [(DetailsModule::class)])
interface DetailsComponent {
    fun inject(target: MovieDetailsFragment)
}
